### ISSUE TITLE
Calling `Close` in another thread crashes

### ISSUE THREAD

**Issue by claunia** [alhimik45/Mono.Fuse.NETStandard#3]
When `Start()` is called in one thread, therefore blocking it, and `Stop()` is called in another thread, `libfuse.so.2` crashes with a SIGSEGV in `fuse_exit()`.

May be similar to https://github.com/fusepy/fusepy/issues/116

**Comment by alhimik45** [comment]
@claunia Yeah. As I understand from FUSE docs `fuse_exit` and thus `Stop()` can be used only inside file system operations.
You can use workaround, as mentioned in fusepy/fusepy#116:
```csharp
using Mono.Unix.Native; // already exists as Mono.Fuse.NETStandard dependency
...
fuseThreadPid = Syscall.getpid();
fs.Start();
...
//in another thread
Syscall.kill(fuseThreadPid, Signum.SIGHUP); //instead fs.Stop ();
```

I'll add informative exception in `Stop()` to clarify this moment.

`fuse_main` that used in `Start()` doesnt allow any in-process stop from another thread, but it is possible to use more low-level calls to implement this. I think I need to implement it as another method like `StartEventLoop()`. And with it `Stop()` will work in expected intuitive manner.
